src/sagemaker_container_support/environment.py

```diff
@@ -0,0 +1,607 @@
+        """The number of cpus available in the current container.
+
+        Returns:
+            (int): number of cpus available in the current container.
+        """
+        return self._num_cpu
+
+    @property
+    def module_name(self):  # type: () -> str
+        """The name of the user provided module.
+
+        Returns:
+            (str): name of the user provided module
+        """
+        return self._module_name
+
+    @property
+    def module_dir(self):  # type: () -> str
+        """The full path location of the user provided module.
+
+        Returns:
+            (str): full path location of the user provided module.
+        """
+        return self._module_dir
+
+    @property
+    def enable_metrics(self):  # type: () -> bool
+        """Whether metrics should be executed in the environment or not.
+
+        Returns:
+            (bool): representing whether metrics should be execute or not.
+        """
+        return self._enable_metrics
+
+    @property
+    def log_level(self):  # type: () -> int
+        """Environment logging level.
+
+        Returns:
+            (int): environment logging level.
+        """
+        return self._log_level
+
+    @classmethod
+    def create(cls, session=None):  # type: (boto3.Session) -> Environment
+        """
+        Returns: an instance of `Environment`
+        """
+        session = session if session else boto3.Session()
+
+        resource_config = read_resource_config()
+        current_host = resource_config['current_host']
+        hosts = resource_config['hosts']
+
+        input_data_config = read_input_data_config()
+
+        sagemaker_hyperparameters, hyperparameters = split_hyperparameters(read_hyperparameters())
+
+        sagemaker_region = sagemaker_hyperparameters.get(REGION_PARAM_NAME, session.region_name)
+
+        os.environ[JOB_NAME_ENV] = sagemaker_hyperparameters[JOB_NAME_PARAM]
+        os.environ[CURRENT_HOST_ENV] = current_host
+        os.environ[REGION_PARAM_NAME.upper()] = sagemaker_region
+
+        return cls(input_dir=INPUT_PATH,
+                   input_config_dir=INPUT_CONFIG_PATH,
+                   model_dir=MODEL_PATH,
+                   output_dir=OUTPUT_PATH,
+                   output_data_dir=OUTPUT_DATA_PATH,
+                   current_host=current_host,
+                   hosts=hosts,
+                   channel_input_dirs={channel: channel_path(channel) for channel in input_data_config},
+                   num_gpu=gpu_count(),
+                   num_cpu=cpu_count(),
+                   hyperparameters=hyperparameters,
+                   resource_config=resource_config,
+                   input_data_config=read_input_data_config(),
+                   module_name=str(sagemaker_hyperparameters.get(PROGRAM_PARAM)),
+                   module_dir=str(sagemaker_hyperparameters.get(SUBMIT_DIR_PARAM)),
+                   enable_metrics=sagemaker_hyperparameters.get(ENABLE_METRICS_PARAM, False),
```

Comment:
what's the status of the in-container metrics? should we deprecate/remove them?

Reply:
Maybe. Let’s tall about this offline.

Reply:
agree, pulling that is off-topic for this PR